### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ig1na**
Can now list rules and have a prompt asking the user if these rules may be applied. For the moment, the prompt is here for testing purpose and does not apply user's choices.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,11 @@
+# Project Filler CLI
 
+## What is it ?
+
+This project aims to help developers to fill most programming projects (JavaScript, TypeScript, Angular, JAVA..) with tools and good practices proposed by Zenika. It is a command line interface that will detect your project's stack and will search for rules to apply corresponding to this stack.
```
Comment (EmmanuelDemey) [inline README.md]:
```suggestion
This project aims to help developers to fill most programming projects (JavaScript, TypeScript, Angular, JAVA..) with tools and good practices proposed by Zenika. It is a command line interface that will detect your project's stack and will search for rules to apply corresponding to these stacks.
```

File: README.md [hunk 2]
```diff
@@ -1 +1,11 @@
+# Project Filler CLI
 
+## What is it ?
```
Comment (EmmanuelDemey) [inline README.md]:
```suggestion
## What is it for ?
```

File: README.md [hunk 3]
```diff
@@ -1 +1,11 @@
...
+
+This project aims to help developers to fill most programming projects (JavaScript, TypeScript, Angular, JAVA..) with tools and good practices proposed by Zenika. It is a command line interface that will detect your project's stack and will search for rules to apply corresponding to this stack.
+
+## How can I use it ?
+
+For the moment, the project is not really usable, but you can try it using the binary release corresponding to your platform. Download the release tarball, decompress it, and run the bin/project-starter-cli binary in your shell like this :
```
Comment (EmmanuelDemey) [inline README.md]: What is the URL of the binary ?
You can add a Getting Started in order to explain how to launch the project from scratch as a developer.

File: package.json [hunk 4]
```diff
@@ -92,8 +95,6 @@
     "trailingComma": "all"
   },
   "husky": {
-    "hooks": {
-      "pre-push": "npm run ci"
-    }
+    "hooks": {}
```
Comment (EmmanuelDemey) [inline package.json]: ??? :D

File: src/file-utils.ts [hunk 5]
```diff
@@ -4,45 +4,32 @@ import * as fs from 'fs';
...
-  static filesExistIn(path: string, filesPaths: string[]): boolean {
-    filesPaths.forEach(filePath => {
-      try {
-        fs.accessSync(path + filePath);
-      } catch (error) {
-        return false;
-      }
+  static filesExistIn(path: string, fileNames: string[]): boolean {
+    let found = false;
+    fileNames.forEach(fileName => {
```
Comment (EmmanuelDemey) [inline src/file-utils.ts]: Should we stop de forEach when we find the file ?

File: src/file-utils.ts [hunk 6]
```diff
@@ -4,45 +4,32 @@ import * as fs from 'fs';
...
+    return found;
   }
 
-  static findFilesRecursively(path: string, fileName: string): boolean {
+  static findFileRecursively(path: string, fileName: string): boolean {
     let found = false;
-    if (fs.lstatSync(path).isFile()) {
+    const fileStat = fs.lstatSync(path);
+    if (fileStat.isFile() || fileStat.isSymbolicLink()) {
       if (path.endsWith(fileName)) {
```
Comment (EmmanuelDemey) [inline src/file-utils.ts]: `return path.endsWith(fileName);`

File: src/rules/husky.ts [hunk 7]
```diff
@@ -38,14 +38,28 @@ export class Husky {
...
+  getDescription(): string {
+    return 'Husky can prevent bad commits or bad push. Please select rules you would like to add.';
+  }
+
+  getPromptType() {
+    return 'checkbox';
+  }
+
+  getChoices() {
+    return [{ name: 'Rule1', value: 1 }, { name: 'Rule2', value: 2 }];
```
Comment (EmmanuelDemey) [inline src/rules/husky.ts]: Just in case... these values have to be changed.

File: package.json (hunk 4 above)
Comment (ig1na) [reply package.json]: Woups

File: src/file-utils.ts (hunk 5 above)
Comment (ig1na) [reply src/file-utils.ts]: No because the method checks if a list of files all exist

File: src/file-utils.ts (hunk 6 above)
Comment (ig1na) [reply src/file-utils.ts]: good one

File: src/rules/husky.ts (hunk 7 above)
Comment (ig1na) [reply src/rules/husky.ts]: Yes I have to update them, it was just for testing purpose

**Pull request merged**